tools/c7n_azure/c7n_azure/query.py

```diff
@@ -43,29 +50,66 @@ def __init__(self, session_factory):
 
     def filter(self, resource_manager, **params):
         m = resource_manager.resource_type
-        enum_op, list_op, extra_args = m.enum_spec
 
-        if extra_args:
-            params.update(extra_args)
+        # Generic Handling
+        if hasattr(m, 'resource_type'):
```

Comment:
we have this notion of sources in aws, and in azure (re graph support), this feels like it would be another type of query/describe source

Reply:
Yeah I was thinking about it but in the end the old source wouldn't function anymore after I remove some SDK's - so I think I'll just end up replacing the existing primary source entirely.

Reply:
that's fair, this will hopefully be simpler, and avoids some serialize/deserialize loops.

Reply:
yeah.. basically resource managers interpret the request for a named source however they want, so can be overridden per resource.

Reply:
the goal towards making this a default sounds good though where its supported makes sense though, and avoids a lot of extraneous serialization/deserialization into the object models.